docs/proposals/kubelet-oci-runtime.md

```diff
@@ -0,0 +1,87 @@
+containers.  One obvious choice is to use systemd as the parent of these containers.  Another option is to run a wrapper
+process and have the wrapper collect the exit code of the container and make it available to the oci-runtime via a named
+pipe.  The oci-runtime will provide plugins for lifecycle management and both of the options mentioned above will be
+made available.  The kubelet will interact with the runtime to control the lifecycle using the runtime API. Instead of
+the declarative SyncPods API, the kubelet will have much finer control over the pod/container lifecycles. 
+
+### Logging
+Logging will work by using files to redirect stdout / stderr of the containers. Other higher level drivers could build
+on top of files. Using files allow using disk quotas, impose Disk IO limits, and also avoid bottlenecks such as a SPOF
+daemons. This approach also let’s Kubelet manage the lifecycle of logs with policies. Whenever there is disk pressure,
+the kubelet can, Rotate log files of existing containers Prioritize logs from the first and last instance of a container
+and delete the rest Delete logs from dead pods & containers.  Impose per-container default Disk IO limits to help fairly
+share local disk.  Provide a common extension point for logging daemons to collect logs and metadata from Kubernetes
+deployments.
+
+### Rootfs management
+The kubelet will be responsible for creating the rootfs for the containers in a pod. Different kinds of root filesystems
+will be supported (such as simple file based, union file systems, read-only with mounts), etc. 
+
+### Networking
+Networking should work just like it works today with hooks running in the network namespace of the infra container. Once
+the pod sandbox has been setup using an infra container, kubelet will let network plugins update networking options in
+the pod sandbox.
+
+### Volumes
+Kubelet volumes will also be supported similar to how they work today using bind mounts in OCI Runtime configuration.
+Security OCI Runtime Specification covers all the security knobs in the linux kernel and is quick to add support for any
+new features that are added to the kernel.
+
+### Container Configuration
+Kubelet runtime configuration will be translated to OCI Specification using Ocitools.
+
+### Resource Management
+OCI Runtime Specification supports cgroups settings as well as joining cgroups specified by a path. That allows the
+kubelet to manage the resources of a pod in a fine grained manner.  Also, there is support for systemd driver for
+cgroups, which provides compatibility with distros reliant on systemd. Oci-runtime will comply with the resource
+specification provided by the kubelet via the runtime API.
+
+### Metrics
+cAdvisor will be extended to provide container and pod metrics for the oci-runtime. The oci-runtime will use cAdvisor
+libraries to collect metrics and make it available to kubelet.
```

Comment:
`built-in` to the `oci-runtime` or `kubelet`? My understanding is that runtimes will provide the metrics that kubelet needs. What runtimes use to get metrics is out of scope for the CRI. 
As for `oci-runtime`, it will link in cAdvisor to collect all the metrics needed to satisfy CRI.